lib/src/iterable_extensions.dart

```diff
@@ -30,23 +30,8 @@ extension IterableExtension<T> on Iterable<T> {
   /// The chosen elements are not in any specific order.
   List<T> sample(int count, [Random? random]) {
     RangeError.checkNotNegative(count, 'count');
-    var iterator = this.iterator;
-    var chosen = <T>[];
-    for (var i = 0; i < count; i++) {
-      if (iterator.moveNext()) {
-        chosen.add(iterator.current);
-      } else {
-        return chosen;
-      }
-    }
-    var index = count;
-    random ??= Random();
-    while (iterator.moveNext()) {
-      index++;
-      var position = random.nextInt(index);
-      if (position < count) chosen[position] = iterator.current;
-    }
-    return chosen;
+    var shuffled = [...this]..shuffle(random ?? Random());
```

Comment:
This implementation requires keeping all the iterable's elements in memory at the same time.
If you do that anyway, the `sample` method doesn't make much sense.
The benefit of `sample` is that you can get *n* elements without using more than O(*n*) space, which makes sense for very long iterables.